src/platforms/hosted/utils.c

```diff
@@ -26,27 +26,11 @@
 #include "general.h"
 
 #include <string.h>
-#include <unistd.h>
-#include <sys/time.h>
 
+#include "timeofday.h"
 #include "timing.h"
 #include "bmp_hosted.h"
 
-#if defined(_WIN32) && !defined(__MINGW32__)
-int vasprintf(char **strp, const char *const fmt, va_list ap)
```

Comment:
We're not opposed to this change, but would appreciate understanding the rationale and possibly having a comment in general.h as to why it's there please. (That is, we're interested in why moving the function rather than just making a decl of it available in general.h)

Reply:
There are two uses of  `vasprintf` that seem to not have been tested on MSVC:

https://github.com/blackmagic-debug/blackmagic/blob/88b132edbe9f63da0c9f9754db98724632ac3b53/src/gdb_packet.c#L331

and 

https://github.com/blackmagic-debug/blackmagic/blob/88b132edbe9f63da0c9f9754db98724632ac3b53/src/gdb_packet.c#L353

Moving it is the only avenue because, as you explained in another thread, this affects the non-hosted part of BMP.

Reply:
Hmm, for non-BMDA (ie, firmware) we should be hitting [`vasiprintf`](https://github.com/mirror/newlib-cygwin/blob/fe2545e9faaf4bf9586f61a7b83d5cb5af501194/newlib/libc/stdio/vasprintf.c#L43) from newlib for this, which is why we were wondering - hence too why the function was only in BMDA's utilities impl. Hence too the guarding on the decl as this is also found in the Cygwin CRT (part of its POSIX compat layer).

Reply:
I meant BMDA under MSVC, not BMDA under MinGW or Cygwin.

Reply:
Ok, that helps with clarity - thank you. Guess that our main question then is why a decl like the following couldn't have been added to general.h to cover this case?:

```c
#if defined(_MSC_VER)
int vasprintf(char **strp, const char *const fmt, va_list ap);
#endif
```

We're interested in what makes this impractical as there feels like a comment or some documentation should be written given that.

Reply:
From a semantic point of view, such a declaration would be making the BMD library export a function that is defined in BMDA, a completely separate binary target.

Reply:
Ahhh, ok - that makes a lot more sense then. In that case, please document that with the function's definition in general.h, noting that we think `#if defined(_WIN32) && !defined(__MINGW32__)` needs to be maintained so that it appears in all appropriate builds.

We would like to write the function out of the code base in the future but at least that'll prevent anyone from thinking the same way and making the same mistake as we nearly did here.